### PR TITLE
fix: Redirect on 401

### PR DESCRIPTION
**Pull request by geido**
### SUMMARY
Redirects the user to the login page when 401s responses are caught. It also passes the `next_url` param which is then used to redirect the user to the original page.

### AFTER

https://user-images.githubusercontent.com/60598000/144454347-c068fd34-7f9c-46d6-b0f8-8d98e6336d1d.mp4

### TESTING INSTRUCTIONS
1. Open any page
2. Delete the cookies or wait for fo the auth to expire
3. Make an action that generates a backend request
4. Observe the behavior

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/packages/superset-ui-core/src/connection/callApi/parseResponse.ts [hunk 1]
```diff
@@ -30,7 +36,8 @@ export default async function parseResponse<T extends ParseMethod = 'json'>(
     : T extends 'text'
     ? TextResponse
     : never;
-  const response = await apiPromise;
+
+  const response = isPromise(apiRequest) ? await apiRequest : apiRequest;
```
Comment (suddjian) [inline superset-frontend/packages/superset-ui-core/src/connection/callApi/parseResponse.ts:40]:
```suggestion
  const response = await Promise.resolve(apiRequest);
```
You don't really need the `isPromise` check

**Review by suddjian — APPROVED** [review]

**Review by kgabryje — APPROVED** [review]
Works great!
1 suggestion for a ux improvement in another PR - we should display a banner on login page that would inform user that their session expired

**Comment by kgabryje** [comment]
I'm not getting redirected back to the original page when I log in

https://user-images.githubusercontent.com/15073128/145011196-4d6c2628-4519-4e2f-b367-371e80ca4c33.mov

**Comment by jinghua-qa** [comment]
🏷️ 2021.49

**Pull request merged**